tests/core/test_scripts.py

```diff
@@ -3287,6 +3287,34 @@ def test_should_process_lane(self, db: DatabaseTransactionFixture):
         worklist = WorkList()
         assert False == script.should_process_lane(worklist)
 
+    def test_site_configuration_has_changed(
+        self,
+        db: DatabaseTransactionFixture,
+        external_search_patch_fixture: ExternalSearchPatchFixture,
+    ):
+        lane1 = db.lane()
+        lane2 = db.lane()
+        lane1.size = 100
+        lane2.size = 50
+
+        # Commit changes to the DB so the lane creation listeners are fired
+        db.session.commit()
+
+        with patch("core.lane.site_configuration_has_changed") as lane_changed:
+            with patch(
+                "core.scripts.site_configuration_has_changed"
+            ) as scripts_changed:
+                UpdateLaneSizeScript(db.session).do_run(cmd_args=[])
```

Comment:
Fun fact: You can write that as a single `with` statement and, thus, add only a single level of indenting. Warning: it looks hideous. 😱

```
with patch(
    "core.lane.site_configuration_has_changed"
) as lane_changed, patch(
    "core.scripts.site_configuration_has_changed"
) as scripts_changed:
    UpdateLaneSizeScript(db.session).do_run(cmd_args=[])
```

You could also use backslashes for continuation to get each context manager on it's own line, but I'm not sure how our linter overlords will feel about that. 😅

Python 3.10 supports [parenthesized context managers](https://docs.python.org/3.10/whatsnew/3.10.html#parenthesized-context-managers), so we might just want to wait until that's our minimum support level before doing this kind of combining. But imagine...

```
with (
    patch("core.lane.site_configuration_has_changed") as lane_changed,
    patch("core.scripts.site_configuration_has_changed") as scripts_changed,
):
    UpdateLaneSizeScript(db.session).do_run(cmd_args=[])
```

Reply:
🙏🏻 Thank you. I didn't know you could use multiple context managers like that.